db_test.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2012 James Cooper. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+// Package gorp provides a simple way to marshal Go structs to and from
+// SQL databases.  It uses the database/sql package, and should work with any
+// compliant database/sql driver.
+//
+// Source code and project home:
+// https://github.com/go-gorp/gorp
+
+package gorp
```

Comment:
Should be `package gorp_test`

There are a lot of reasons for this, but the biggest, most important one is: when you test unexported functionality, you're not testing a use case of the code.  In this case, the tests that are in this package don't actually prove that `gorp` works as expected from a user's perspective.  They prove that `expandSliceArgs` works as expected, but they don't prove that `gorp` ever calls `expandSliceArgs`.

I know testing exported functionality only is more difficult and frustrating (especially in a project like gorp where all of our tests are integration tests), but it's much safer _and_ gives the end user a better example of how they might use it.

Reply:
Sure! I will work on that.

Reply:
thanks! :)

Reply:
✅ Done! Changed the test to only access an exported method.

Should I test all exported methods that I added the `expandSliceArgs` call? Because they would all call the same piece of code.

Reply:
@nelsam Got the error bellow on Travis built, related to a `fmt.Errorf` message that I didn't work on. Looking at [this line](https://github.com/go-gorp/gorp/blob/6032c66e0f5f155fd56216ed14cbbdd991034605/select.go#L134) it does not appear that we need to add `...` to `args`. Any clues for that? 😄 

```
# github.com/go-gorp/gorp
./select.go:134: missing ... in args forwarded to printf-like function
FAIL	github.com/go-gorp/gorp [build failed]
```

Reply:
That failure only seemed to happen in go `tip`, which is an allowed failure - you can ignore it, I think.

The real failure is in [the test run with postgres](https://travis-ci.org/go-gorp/gorp/jobs/394025421#L4715-L4720):
```
--- FAIL: TestDbMap_Select_expandSliceArgs (0.01s)
    --- FAIL: TestDbMap_Select_expandSliceArgs/it_should_handle_slice_placeholders_correctly (0.00s)
    	db_test.go:180: pq: column "field1" does not exist
    --- FAIL: TestDbMap_Select_expandSliceArgs/it_should_handle_slice_placeholders_correctly_with_custom_types (0.00s)
    	db_test.go:180: pq: column "field2" does not exist
```

Reply:
Thanks @nelsam , solved the postgres issue. `Go 1.9` and `Go 1.10` tests are green.

Reply:
cool; I'll review after work